File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n=nums.size();
        int left=0;
        int mid=0;
        int right=n-1;
        while(mid<=right){
            if(nums[mid]==0){
                swap(nums[left], nums[mid]);
                left++;
                mid++;
            }
            else if(nums[mid]==1){
                mid++;
            }
            else{
                swap(nums[mid], nums[right]);
                right--;
            }

        }
    }
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n=len(nums)
        left=0
        mid=0
        right=n-1

        while mid<=right:
            if nums[mid]==1:
                mid+=1
            elif nums[mid]==0:
                nums[left], nums[mid] = nums[mid], nums[left]
                left+=1
                mid+=1
            else:
                nums[right], nums[mid] = nums[mid], nums[right]
                right-=1
        
        # return nums